### PR TITLE
figure1 fix and requirements freeze

### PR DESCRIPTION
**Pull request by zachschuermann**
Added a quick fix for figure1:  
I think we were trying to get the first entry like this
```
ret_json['result']['course name'] = str(df['Course Title'][0])
...
```
but just changed to this to do integer indexing
```
ret_json['result']['course name'] = str(df['Course Title'].iloc[0])
...
```

also froze requirements so we can make sure everything stays reproducible

### CODE REVIEW

**Review by samjett247 — CHANGES_REQUESTED** [review]
Zach, could you do me a favor? Can you pull down the most recent commit to this branch then run `python data_loader.py`? That should take the responses and questions that joe scraped and incorporate those into the Mongo `reviews_db`.

File: server.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 CORS(app)
 # useful for testing
 # curl -i http://localhost:5050/api/v0/
```
Comment (samjett247) [inline server.py:27]: Does it serve at port 5050? If not, delete

Comment (zachschuermann) [reply server.py:27]: actually I'm going to change it back to the development server running on 5050 ok?

**Comment by zachschuermann** [comment]
data_loader failed, here is output:

```
(env) Zach-MacBook:stev-api Zach$ python data_loader.py 
Converting the file GCOE.csv to pd dataframe.
Converting the file JRCOE.csv to pd dataframe.
Converting the file COAS.csv to pd dataframe.
Converting the scraped collection ARC to pd dataframe.
Converting the scraped collection BUS to pd dataframe.
Converting the scraped collection FARTS to pd dataframe.
Converting the scraped collection GEO to pd dataframe.
Converting the scraped collection INTS to pd dataframe.
Converting the scraped collection JRNL to pd dataframe.
Converting the scraped collection NRG to pd dataframe.
Loading GCOE
A collection called GCOE was added to the database reviews-db.
Aggregating the GCOE. This usually takes about a minute.
100%|███████████████████████████████████████████████████████████████████████████████████| 25/25 [01:32<00:00,  4.61s/it]
0 entries were dropped from main dataset, corresponding to 0 dropped from the aggregated dataframe.


A collection called aggregated_GCOE was added to the database reviews-db.
Loading JRCOE
A collection called JRCOE was added to the database reviews-db.
Aggregating the JRCOE. This usually takes about a minute.
100%|███████████████████████████████████████████████████████████████████████████████████| 25/25 [01:54<00:00,  4.29s/it]
0 entries were dropped from main dataset, corresponding to 0 dropped from the aggregated dataframe.


A collection called aggregated_JRCOE was added to the database reviews-db.
Loading COAS
A collection called COAS was added to the database reviews-db.
Aggregating the COAS. This usually takes about a minute.
100%|███████████████████████████████████████████████████████████████████████████████████| 25/25 [21:51<00:00, 57.69s/it]
0 entries were dropped from main dataset, corresponding to 0 dropped from the aggregated dataframe.


A collection called aggregated_COAS was added to the database reviews-db.
Loading ARC
A collection called ARC was added to the database reviews-db.
Aggregating the ARC. This usually takes about a minute.
  0%|                                                                                                                                                               | 0/12 [00:00<?, ?it/s]Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: |D, course: 1253, and instructor: 1954625223
     College Code  Course Number  Department Responses  Avg Department Rating  SD Department Rating  ...  Course Title Course Enrollment  Subject Code  Term Code  Instructor Enrollment
0             ARC           1253                   4.0                    0.0                   0.0  ...   Graphics II                 0            |D     201410                    4.0
4065          ARC           1253                   6.0                    0.0                   0.0  ...   Graphics II                 0            |D     201410                    6.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: |D, and number: 1253 in term201410
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: |D, course: 2773, and instructor: 1594081673
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
1240          ARC           2773                  26.0                    0.0  ...                0.0            |D    201410                   26.0
1242          ARC           2773                  25.0                    0.0  ...                0.0            |D    201410                   25.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: |D, and number: 2773 in term201410
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: |D, course: 2543, and instructor: 1744544966
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
3310          ARC           2543                  18.0                    0.0  ...                0.0            |D    201410                   18.0
3313          ARC           2543                  17.0                    0.0  ...                0.0            |D    201410                   17.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: |D, and number: 2543 in term201410
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: ARCH, course: 4233, and instructor: 916451911
   College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
96          ARC           4233                  22.0                    0.0  ...                0.0          ARCH    201410                   22.0
99          ARC           4233                  21.0                    0.0  ...                0.0          ARCH    201410                   21.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: ARCH, and number: 4233 in term201410
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: ARCH, course: 4970, and instructor: 1302930978
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
1120          ARC           4970                   3.0                    0.0  ...                0.0          ARCH    201410                    3.0
3380          ARC           4970                   2.0                    0.0  ...                0.0          ARCH    201410                    2.0

[2 rows x 19 columns]


Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: ARCH, course: 2454, and instructor: 159880501
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
2795          ARC           2454                  11.0                    0.0  ...                0.0          ARCH    201410                   11.0
2796          ARC           2454                  10.0                    0.0  ...                0.0          ARCH    201410                   10.0

[2 rows x 19 columns]


Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: ARCH, course: 4543, and instructor: 57154276
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
2900          ARC           4543                   7.0                    0.0  ...                0.0          ARCH    201410                    7.0
2904          ARC           4543                   6.0                    0.0  ...                0.0          ARCH    201410                    6.0

[2 rows x 19 columns]


Aggregated Dataframe contains incorrect number of entries (number entries = 3) for term201410, subject: ARCH, course: 4623, and instructor: 1857834420
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
3605          ARC           4623                   1.0                    0.0  ...                0.0          ARCH    201410                    1.0
3606          ARC           4623                  14.0                    0.0  ...                0.0          ARCH    201410                   14.0
3607          ARC           4623                  15.0                    0.0  ...                0.0          ARCH    201410                   15.0

[3 rows x 19 columns]


Error: forced to eliminate the course with subject/department: ARCH, and number: 4623 in term201410
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: LA, course: 4970, and instructor: 1324493823
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
3225          ARC           4970                   6.0                    0.0  ...                0.0            LA    201410                    6.0
3345          ARC           4970                   3.0                    0.0  ...                0.0            LA    201410                    3.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: LA, and number: 4970 in term201410
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: CNS, course: 2822, and instructor: 1998108427
    College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
625          ARC           2822                  17.0                    0.0  ...                0.0           CNS    201410                   17.0
626          ARC           2822                  16.0                    0.0  ...                0.0           CNS    201410                   16.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: CNS, and number: 2822 in term201410
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201410, subject: CNS, course: 4153, and instructor: 331440457
     College Code  Course Number  Department Responses  Avg Department Rating  SD Department Rating  ...         Course Title Course Enrollment  Subject Code  Term Code  Instructor Enrollment
4045          ARC           4153                  10.0                    0.0                   0.0  ...  Legal Issues-Constr               0.0           CNS     201410                   10.0
4046          ARC           4153                   9.0                    0.0                   0.0  ...  Legal Issues-Constr               0.0           CNS     201410                    9.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: CNS, and number: 4153 in term201410
  8%|████████████▌                                                                                                                                          | 1/12 [00:02<00:29,  2.71s/it]Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201610, subject: CNS, course: 2812, and instructor: 922785590
  College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
5          ARC           2812                  13.0                    0.0  ...                0.0           CNS    201610                   13.0
8          ARC           2812                  12.0                    0.0  ...                0.0           CNS    201610                   12.0

[2 rows x 19 columns]


Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201610, subject: CNS, course: 2812, and instructor: 1954625223
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
2170          ARC           2812                  14.0                    0.0  ...                0.0           CNS    201610                   14.0
2173          ARC           2812                  13.0                    0.0  ...                0.0           CNS    201610                   13.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: CNS, and number: 2812 in term201610
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201610, subject: CNS, course: 4993, and instructor: 588405501
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
1103          ARC           4993                   7.0                    0.0  ...                0.0           CNS    201610                    7.0
4095          ARC           4993                   8.0                    0.0  ...                0.0           CNS    201610                    8.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: CNS, and number: 4993 in term201610
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201610, subject: CNS, course: 5303, and instructor: 1847094773
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
1145          ARC           5303                   6.0                    0.0  ...                0.0           CNS    201610                    6.0
1148          ARC           5303                   5.0                    0.0  ...                0.0           CNS    201610                    5.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: CNS, and number: 5303 in term201610
Aggregated Dataframe contains incorrect number of entries (number entries = 2) for term201610, subject: CNS, course: 3823, and instructor: 1998108427
     College Code  Course Number  Department Responses  Avg Department Rating  ...  Course Enrollment  Subject Code Term Code  Instructor Enrollment
1930          ARC           3823                  11.0                    0.0  ...                0.0           CNS    201610                   11.0
1932          ARC           3823                   1.0                    0.0  ...                0.0           CNS    201610                    1.0

[2 rows x 19 columns]


Error: forced to eliminate the course with subject/department: CNS, and number: 3823 in term201610
/Users/Zach/Desktop/Dev/STEV/stev-api/data_aggregation.py:32: RuntimeWarning: invalid value encountered in double_scalars
  mean = np.sum(mean_list*pop_list*weight_list)/(np.sum(pop_list*weight_list))
Nan for instructor mean
2025    5.0
2026    5.0
2027    5.0
2028    5.0
2029    5.0
Name: Mean, dtype: float64
2025    1.0
2026    1.0
2027    1.0
2028    1.0
2029    1.0
Name: Standard Deviation, dtype: float64
2025    0.0
2026    0.0
2027    0.0
2028    0.0
2029    0.0
Name: Responses, dtype: float64
2025    1
2026    1
2027    1
2028    1
2029    1
Name: Question Number, dtype: int64
Nan for (number entries = 1) for term201610, subject: CNS, course: 4960, and instructor: 1998108427

Traceback (most recent call last):
  File "data_loader.py", line 141, in <module>
    update_database(force_update=True)
  File "data_loader.py", line 122, in update_database
    ag_records = json.loads(ag_df.T.to_json()).values()
AttributeError: 'int' object has no attribute 'T'
```

**Comment by zachschuermann** [comment]
All good now

**Review by samjett247 — APPROVED** [review]
Looks good to me. Thanks for the work on this zach; sorry about the trouble

**Pull request merged**